logic/transactions/pool.js

```diff
@@ -0,0 +1,705 @@
+	self.bundledInterval = library.config.broadcasts.broadcastInterval;
+	pool = {
+		unverified: { transactions: [], index: {} },
+		verified:{
+			pending: { transactions: [], index: {} },
+			ready: { transactions: [], index: {} }
+		}
+	};
+
+	// Bundled transaction timer
+	function nextBundle (cb) {
+		self.processPool(function (err) {
+			if (err) {
+				library.logger.log('processPool transaction timer', err);
+			}
+			return setImmediate(cb);
+		});
+	}
+
+	jobsQueue.register('txPoolNextBundle', nextBundle, self.bundledInterval);
+
+}
+
+// Public methods
+/**
+ * Bounds input parameters to private variable modules.
+ * @param {Accounts} accounts
+ * @param {Transactions} transactions
+ * @param {Loader} loader
+ */
+TxPool.prototype.bind = function (accounts, transactions, loader) {
+	modules = {
+		accounts: accounts,
+		transactions: transactions,
+		loader: loader,
+	};
+};
+
+/**
+ * Gets unverified, verified.pending and verified.ready indexes and transactions length.
+ * @return {Object} unverified, pending, ready
+ */
+TxPool.prototype.getUsage = function () {
+	return {
+		unverified: {
+			indexes: Object.keys(pool.unverified.index).length,
+			txs: pool.unverified.transactions.length
+		},
+		pending: {
+			indexes: Object.keys(pool.verified.pending.index).length,
+			txs: pool.verified.pending.transactions.length
+		},
+		ready: {
+			indexes: Object.keys(pool.verified.ready.index).length,
+			txs: pool.verified.ready.transactions.length
+		}
+	};
+};
+
+/**
+ * Gets transaction based on transaction id.
+ * Checks all pool lists: unverified, pending, ready.
+ * @param {string} id
+ * @return {Object} tx, status
+ */
+TxPool.prototype.get = function (id) {
+	var index = pool.unverified.index[id];
+	if (index !== undefined) {
+		return {
+			tx: pool.unverified.transactions[index],
+			status: 'unverified'
+		};
+	}
+
+	index = pool.verified.pending.index[id];
+	if (index !== undefined) {
+		return {
+			tx: pool.verified.pending.transactions[index],
+			status: 'pending'
+		};
+	}
+	
+	index = pool.verified.ready.index[id];
+	if (index !== undefined) {
+		return {
+			tx: pool.verified.ready.transactions[index],
+			status: 'ready'
+		};
+	}
+	return {
+		tx: undefined,
+		status: 'Not in pool'
+	};
+};
+
+/**
+ * Gets ready transactions ordered by fee and received time.
+ * @param {number} limit
+ * @return {transaction[]}
+ */
+TxPool.prototype.getReady = function (limit) {
+	var r = _.orderBy(pool.verified.ready.transactions, ['fee', 'receivedAt'],['desc', 'desc']);
+	if (limit && limit < r.length) {
+		r.splice(limit);
+	}
+	return r;
+};
+
+/**
+ * Gets transactions based on limit and pool list.
+ * @implements {__private.getTxsFromPoolList}
+ * @param {string} poolList unverified, pending, ready
+ * @param {boolean} reverse
+ * @param {number} limit
+ * @return {[transactins]} transactions in pool list filter by limit.
+ */
+TxPool.prototype.getByPoolList  = function (poolList, reverse, limit) {
```

Comment:
just `list` instead of `poolList` or maybe `target`, also you can define function that will return you reference to list instead of doing `switch` all the time
also here `poolList` is a string and in other function `poolList` is object - that is inconsistent